### PR TITLE
Adds grouping to photo vis

### PR DESCRIPTION
**Pull request by asalani93**
Addresses #2187.

You can now group and filter out photos based on what data set they are in.

### CODE REVIEW

**Comment by jaypoulz** [comment]
Out html checker is down again. I though I'd fixed this, but apparently not.

**Comment by bdonald25** [comment]
You should hide groups that don't contain any photos.

File: app/assets/javascripts/templates/visualizations/photo/pic.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!-- Handlebars template for the photo box in the photo vis -->
-<div class="col-xs-6 col-lg-3">
+<div class="col-xs-6 col-lg-3 pic-box">
```
Comment (bdonald25) [inline app/assets/javascripts/templates/visualizations/photo/pic.hbs]: You should add col-md-4.

**Comment by bdonald25** [comment]
Testers: Upload images to different data sets. When you visualize them together you should be able to group by data set. Make sure this works as expected.

File: app/assets/javascripts/visualizations/highvis/photos.coffee [hunk 2]
```diff
@@ -52,39 +46,69 @@ $ ->
...
-                $(canvas).append lbTemp(ctx)
-
-                $('#target-img').modal
-                  keyboard: true
-                $('#target-img').on 'hidden.bs.modal', ->
-                  $('#target-img').remove()
-
-              id++
+        for group in selectedGroups
+          if groupedMedia[group[1]].length == 0
```
Comment (jaypoulz) [inline app/assets/javascripts/visualizations/highvis/photos.coffee]: What if groupedMedia[group[1]] is undefined? (this is occurring in my localhost)

File: app/assets/javascripts/visualizations/highvis/photos.coffee [hunk 3]
```diff
@@ -52,39 +46,69 @@ $ ->
...
+            continue
+
+          groupContext =
+            g_id: group[0]
+            group_label: group[1]
+          groupObj = $(grpTemp(groupContext))
+
+          # put the photos in their correct groups
+          for photo in groupedMedia[group[1]]
```
Comment (jaypoulz) [inline app/assets/javascripts/visualizations/highvis/photos.coffee]: Again, groupedMedia[group[1]] may not exist

**Comment by jaypoulz** [comment]
We should verify that projects with pre-existing photos are not broken, as is the case on my local host.

**Comment by jaypoulz** [comment]
Also, let's try to be more conscious of nullity checks in our future vis patches - it's better for vis to skip draw a small section than to abort mid-draw

**Pull request merged**